src/app/search/search.component.ts

```diff
@@ -35,28 +45,35 @@ export class SearchComponent implements OnInit {
     const searchParams = {};
 
     this.caseFilterFG = filter.formGroup;
-    const caseFilters = this.getCaseFilterFromFormGroup(filter.formGroup);
+    this.metaDataFields = filter.metadataFields;
 
     if (filter.caseState) {
       paginationParams['state'] = filter.caseState.id;
       searchParams['state'] = filter.caseState.id;
     }
 
+    if (filter.page) {
+      searchParams['page'] = filter.page;
+    }
+
+    const filters = this.getCaseFilterFromFormGroup(filter.formGroup);
```

Comment:
I think we need to do some refactoring of this search component sooner or later. I feels like things are getting complicated. 
I think all this code to retrieve filters, caseFilters, metadataFilters, metadataPaginationParams, belongs to the filter itself. But I guess we can't easily move it there because filter here is an untyped param :(